apps/dotcom/sync-worker/src/TLAppDurableObject.ts

```diff
@@ -261,4 +281,19 @@ export class TLAppDurableObject {
 			console.error('failed to persist', e)
 		}
 	}
+
+	async getFileShareType(
+		fileId: TldrawAppFileId,
+		userId: string
+	): Promise<'private' | 'view' | 'edit'> {
+		if (!this._userId) {
+			this._userId = userId
+		} else if (userId !== this._userId) {
+			throw new Error('userId mismatch')
+		}
+		const room = await this.getRoom()
+		const file = room.getRecord(fileId) as TldrawAppFile | undefined
+		if (!file?.shared) return 'private'
```

Comment:
later on, when we have more granular sharing controls (i.e. per-user controls) is this going to be able to be refactored easily?

Reply:
Sure